lib/hci-socket/gatt.js

```diff
@@ -72,175 +105,254 @@ var Gatt = function() {
 
 util.inherits(Gatt, events.EventEmitter);
 
-Gatt.prototype.setServices = function(services) {
-  var deviceName = process.env.BLENO_DEVICE_NAME || os.hostname();
+function createATTMessage(op, mtu, valueHandle, data){
+    var dataLength = Math.min(data.length, mtu - 3);
+    var message = new Buffer(3 + dataLength);
 
-  // base services and characteristics
-  var allServices = [
-    {
-      uuid: '1800',
-      characteristics: [
-        {
-          uuid: '2a00',
-          properties: ['read'],
-          secure: [],
-          value: new Buffer(deviceName),
-          descriptors: []
-        },
-        {
-          uuid: '2a01',
-          properties: ['read'],
-          secure: [],
-          value: new Buffer([0x80, 0x00]),
-          descriptors: []
-        }
-      ]
-    },
-    {
-      uuid: '1801',
-      characteristics: [
-        {
-          uuid: '2a05',
-          properties: ['indicate'],
-          secure: [],
-          value: new Buffer([0x00, 0x00, 0x00, 0x00]),
-          descriptors: []
-        }
-      ]
+    message.writeUInt8(parseInt(op, 16), 0);
+    message.writeUInt16LE(parseInt(valueHandle, 16), 1);
+
+    for (i = 0; i < dataLength; i++) {
+      message[3 + i] = data[i];
     }
-  ].concat(services);
 
-  this._handles = [];
+    return message;
+}
 
-  var handle = 0;
-  var i;
-  var j;
+function createServiceHandles(service, start){
+  var handle = start;
+  var handles = [];
 
-  for (i = 0; i < allServices.length; i++) {
-    var service = allServices[i];
+  handle++;
```

Comment:
Similarly, the handle can't be ever higher than 0xFFFF, in your updated pizza example, swapping every 1.5 seconds, it will eventually exceed that.

Need to think about a handle management strategy ...